app/src/main/java/ai/kun/opentrace/ui/home/HomeFragment.kt

```diff
@@ -1,31 +1,233 @@
 package ai.kun.opentrace.ui.home
 
+import ai.kun.opentrace.R
+import ai.kun.opentrace.worker.BLETrace
+import android.Manifest
+import android.bluetooth.BluetoothAdapter
+import android.bluetooth.BluetoothManager
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
 import android.os.Bundle
+import android.util.Log
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.TextView
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProviders
-import ai.kun.opentrace.R
 
 class HomeFragment : Fragment() {
+    private val TAG = "HomeFragment"
 
     private lateinit var homeViewModel: HomeViewModel
 
+    private val REQUEST_ENABLE_BT = 1
+    private val REQUEST_FINE_LOCATION = 2
+    private val REQUEST_BACKGROUND_LOCATION = 3
+
+    private var mIsForeground = true
```

Comment:
This applies to all variables naming. Putting `m` before a name is Java-style convention, not Kotlin one.
See more here:
https://developer.android.com/kotlin/style-guide

Reply:
I follow the java convention on the private stuff, but the kotlin one on public stuff.  I guess I should fully drink the Kotlin Koolaid.